sdk/src/server-api/sc/api/plugins/IGameInstance.kt

```diff
@@ -0,0 +1,87 @@
+package sc.api.plugins
+
+import sc.api.plugins.exceptions.GameLogicException
+import sc.api.plugins.exceptions.TooManyPlayersException
+import sc.api.plugins.host.IGameListener
+import sc.framework.plugins.Player
+import sc.protocol.responses.ProtocolMessage
+import sc.shared.InvalidMoveException
+import sc.shared.PlayerScore
+import sc.shared.ScoreCause
+import kotlin.jvm.Throws
+
+interface IGameInstance {
+    /**
+     * @return the player that joined
+     *
+     * @throws TooManyPlayersException when game is already full
+     */
+    @Throws(TooManyPlayersException::class)
+    fun onPlayerJoined(): Player
+    fun onPlayerLeft(player: Player, cause: ScoreCause? = null)
+
+    /**
+     * Called by the Server once an action was received.
+     *
+     * @param fromPlayer The player who invoked this action.
+     * @param data       ProtocolMessage with the action
+     *
+     * @throws GameLogicException   if any invalid action is done
+     * @throws InvalidMoveException if the received move violates the rules
+     */
+    @Throws(GameLogicException::class, InvalidMoveException::class)
+    fun onAction(fromPlayer: Player, data: ProtocolMessage)
+    fun addGameListener(listener: IGameListener)
+    fun removeGameListener(listener: IGameListener)
+
+    /** Server or an administrator requests the game to start now.  */
+    fun start()
+
+    /**
+     * Destroys the Game.
+     * Might be invoked by the server at any time. Any open handles should be removed.
+     * No events (GameOver etc) should be sent out after this method has been called.
+     */
+    fun destroy()
+
+    /**
+     * The game is requested to load itself from a file (the board i.e.).
+     * Similar to a replay but with actual clients.
+     *
+     * @param file File the game should be loaded from
+     */
+    fun loadFromFile(file: String)
+
+    /**
+     * The game is requested to load itself from a file (the board i.e.).
```

Comment:
"i.e." is always prefixed from my knowledge. Here you might as well omit it.

Reply:
I'd guess this was supposed to be "etc." or an "e.g." prefix

Reply:
removed in 2981bd5a because I don't see it making the description clearer